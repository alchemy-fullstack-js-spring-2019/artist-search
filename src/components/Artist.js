import React from 'react';
import PropTypes from 'prop-types';

function Artist({ artist }){
  return (
    <p>{artist.name} - {artist.disambiguation}</p>
  );
}

Artist.propTypes = {
  artist:PropTypes.object.isRequired
};

export default Artist;
