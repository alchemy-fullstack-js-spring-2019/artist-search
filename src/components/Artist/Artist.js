import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

function Artist({ name, id }) {
  return (
    <p>
      <Link to={ `/${name}/${id}` }>{ name }</Link>
    </p>
  );
}

Artist.propTypes = {
  name: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired
};

export default Artist;
