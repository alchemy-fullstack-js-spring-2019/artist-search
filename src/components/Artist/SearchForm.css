@import url('https://fonts.googleapis.com/css?family=Titillium+Web&display=swap');

form {
  button{
    background-color: skyblue;
    border-radius: 5px;
    width: 8vw;
    height: 3vh;
    margin: 2vh;
  }

}
