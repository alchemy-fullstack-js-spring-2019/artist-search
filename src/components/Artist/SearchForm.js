import React from 'react';
import PropTypes from 'prop-types';
import styles from './SearchForm.css';


function SearchForm({ search, handleChange, searchSubmit }) {
  return (
    <form style={styles} onSubmit={ searchSubmit }>
      <input name="search" type="text" onChange={ handleChange } value={ search } /><br/>
      <button type="submit">Search</button>
    </form>
  );
}

SearchForm.propTypes = {
  search: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired,
  searchSubmit: PropTypes.func.isRequired
};

export default SearchForm;
