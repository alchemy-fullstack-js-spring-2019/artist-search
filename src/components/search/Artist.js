import React from 'react';
import PropTypes from 'prop-types';
import {
  Link
} from 'react-router-dom';

function Artist({ artist, id }) {
  return <Link to={`/artist/${id}/${artist}`}>{artist}</Link>;
}

Artist.propTypes = {
  artist: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired
};

export default Artist;
