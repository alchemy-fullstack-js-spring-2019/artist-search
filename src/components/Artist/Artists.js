import React from 'react';
import PropTypes from 'prop-types';
import Artist from './Artist';

function Artists({ artistArray }) {
  const artistList = artistArray.map(artist => {
    return (
      <li key={ artist.id }>
        <Artist id={ artist.id } name={ artist.name } />
      </li>
    );
  });

  return (
    <ul>
      { artistList }
    </ul>
  );
}

Artists.propTypes = {
  artistArray: PropTypes.array.isRequired
};

export default Artists;
