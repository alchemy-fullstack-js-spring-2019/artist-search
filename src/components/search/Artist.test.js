import React from 'react';
import { shallow } from 'enzyme';
import Artist from './Artist';

describe('Artist component', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<Artist artist='' id='' />);
    expect(wrapper).toMatchSnapshot();
  });
});
