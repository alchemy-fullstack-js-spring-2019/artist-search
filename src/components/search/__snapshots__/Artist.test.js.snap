// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artist component matches snapshot 1`] = `
<Link
  to="/artist//"
/>
`;
